src/components/elements/StepDialog.tsx

```diff
@@ -0,0 +1,136 @@
+import {
+  Box,
+  Button,
+  ButtonProps,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  Tab,
+} from '@mui/material';
+import { ArrowLeftIcon, ArrowRightIcon } from '@mui/x-date-pickers';
+import {
+  ReactNode,
+  SyntheticEvent,
+  useCallback,
+  useMemo,
+  useState,
+} from 'react';
+import CommonDialog, {
+  CommonDialogProps,
+} from '@/components/elements/CommonDialog';
+import Loading from '@/components/elements/Loading';
+import FormDialogActionContent, {
+  FormDialogActionProps,
+} from '@/modules/form/components/FormDialogActionContent';
+
+export type TabDefinition = {
+  title: string;
+  content: ReactNode;
+  FormDialogActionProps?: Partial<FormDialogActionProps>;
+  disableSubmit?: boolean;
+};
+
+interface Props extends Omit<CommonDialogProps, 'onSubmit' | 'onClose'> {
+  title: string;
+  tabDefinitions: TabDefinition[];
+  submitButtonTitle?: string;
+  SubmitButtonProps?: ButtonProps;
+  successContent?: ReactNode;
+  onSubmit: VoidFunction;
+  onClose: VoidFunction;
+  loading?: boolean;
+}
+
+const StepDialog = ({
+  title,
+  submitButtonTitle,
+  SubmitButtonProps,
+  successContent,
+  tabDefinitions,
+  onSubmit,
+  onClose,
+  loading,
+  ...rest
+}: Props) => {
+  const [tabValue, setTabValue] = useState(tabDefinitions[0].title);
+
+  const handleChange = useCallback(
+    (event: SyntheticEvent, newValue: string) => {
+      setTabValue(newValue);
+    },
+    []
+  );
+
+  const [previousTab, currentTab, nextTab] = useMemo(() => {
+    const currentTabIndex = tabDefinitions.findIndex(
+      (t) => t.title === tabValue
+    );
+
+    return [
+      tabDefinitions[currentTabIndex - 1],
+      tabDefinitions[currentTabIndex],
+      tabDefinitions[currentTabIndex + 1],
+    ];
+  }, [tabDefinitions, tabValue]);
+
+  const handleSubmit = useCallback(() => {
+    if (nextTab) {
+      setTabValue(nextTab.title);
+    } else {
+      onSubmit();
+    }
+  }, [nextTab, onSubmit]);
+
+  return (
+    <CommonDialog onClose={onClose} {...rest}>
+      <DialogTitle>{title}</DialogTitle>
+      <DialogContent>
+        {loading && <Loading />}
+        {!loading && !successContent && (
+          <TabContext value={tabValue}>
+            <Box sx={{ borderBottom: 1, borderColor: 'divider' }}>
+              <TabList onChange={handleChange} aria-label={`${title} steps`}>
+                {tabDefinitions.map((tab: TabDefinition) => (
+                  <Tab key={tab.title} label={tab.title} value={tab.title} />
+                ))}
+              </TabList>
+            </Box>
+            {tabDefinitions.map((tab: TabDefinition) => (
+              <TabPanel key={tab.title} value={tab.title}>
+                {tab.content}
+              </TabPanel>
+            ))}
+          </TabContext>
+        )}
+        {successContent && <Box mt={2}>{successContent}</Box>}
+      </DialogContent>
+      {!successContent && (
+        <DialogActions>
+          <FormDialogActionContent
```

Comment:
To discuss:
It doesn't seem like `FormDialogActionContent` is a great fit for this dialog. It has a pretty specific structure geared around dialogs that contain a single submittable form.

I wonder if it would make more sense to introduce a separate component for specifying DialogActions for a workflow stepper dialog like this one.

Reply:
For example: here we are overloading the `otherActions` prop to hold the "back" button in the workflow. For normal form dialogs, I believe we always use `otherActions` for rendering the delete button. If we want to fiddle with where the delete button lives for form dialogs, then we're also messing with where the back button is on workflow step dialogs, etc.